File: backend/src/main/java/com/bookstore/backend/business/requests/author/CreateAuthorRequest.java
package com.bookstore.backend.business.requests.author;


import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class CreateAuthorRequest {

    @NotNull(message = "Name field cannot be null")
    @NotBlank
    @Size(min = 5)
    private String name;

}

File: backend/src/main/java/com/bookstore/backend/dto/requests/author/CreateAuthorRequest.kt
package com.bookstore.backend.dto.requests.author

import javax.validation.constraints.Min
import javax.validation.constraints.NotBlank

data class CreateAuthorRequest(
        @field:NotBlank(message = "Name field cannot be empty")
        @field:Min(4, message = "Name field must be minimum 4 characters")
        val name: String?
)
